src/Streamly/Coreutils/Ln.hs

```diff
@@ -0,0 +1,51 @@
+-- |
+-- Module      : Streamly.Coreutils.Ln
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- create a link to TARGET with the given name
+
+module Streamly.Coreutils.Ln
+    ( ln
+    , setForce
+    , setSymbolic
+    )
+where
+
+import Data.Default.Class (Default(..))
+import qualified System.Posix.Files as Posix
+import Streamly.Coreutils.Common (Switch(..))
+
+data Ln = Ln
+    { force :: Switch
+    , symbolic :: Switch
+    }
+
+instance Default Ln where
+    def = Ln Off Off
+
+setForce :: Switch -> Ln -> Ln
+setForce opt ln0 = ln0 {force = opt}
+
+setSymbolic :: Switch -> Ln -> Ln
+setSymbolic opt ln0 = ln0 {symbolic = opt}
+
+ln :: Ln -> FilePath -> FilePath -> IO ()
+ln opt src tgt =
+    case symbolic opt of
+        Off -> act Posix.createLink
+        On -> act Posix.createSymbolicLink
+
+    where
+
+    act f =
+        case force opt of
+            Off -> do
+                exists <- Posix.fileExist tgt
+                if exists
+                then error "Target exists"
```

Comment:
We can prefix "ln:" with the message so that we know where it is coming from when this comes from a program. Also make it a bit clearer like "ln: Link target exists, use force to create anyway".